java/flight/flight-core/src/test/java/org/apache/arrow/flight/client/TestCookieHandling.java

```diff
@@ -29,37 +45,47 @@
  */
 public class TestCookieHandling {
   private static final String SET_COOKIE_HEADER = "Set-Cookie";
-  private static final String SET_COOKIE2_HEADER = "Set-Cookie2";
-
-  private ClientCookieMiddleware cookieMiddleware = new ClientCookieMiddleware();
+  private static final String COOKIE_HEADER = "Cookie";
+  private BufferAllocator allocator;
+  private FlightServer server;
+  private FlightClient client;
+
+  private ClientCookieMiddleware.Factory factory = new ClientCookieMiddleware.Factory();
+  private ClientCookieMiddleware cookieMiddleware = new ClientCookieMiddleware(factory);
+
+  @Before
+  public void setup() throws Exception {
+    allocator = new RootAllocator(Long.MAX_VALUE);
```

Comment:
You must clean up the allocator, server, and client.

Reply:
Done.